crates/ra_project_model/src/workspace_error.rs

```diff
@@ -0,0 +1,57 @@
+//! Workspace-related errors
+
+use std::{error::Error, fmt, io, path::PathBuf, string::FromUtf8Error};
+
+use ra_db::ParseEditionError;
+
+#[derive(Debug)]
+pub enum WorkspaceError {
+    CargoMetadataFailed(cargo_metadata::Error),
+    CargoTomlNotFound(PathBuf),
+    NoStdLib(PathBuf),
+    OpenWorkspaceError(io::Error),
+    ParseEditionError(ParseEditionError),
+    ReadWorkspaceError(serde_json::Error),
+    RustcCfgError,
+    RustcError(io::Error),
+    RustcOutputError(FromUtf8Error),
+    SysrootNotFound,
+}
```

Comment:
As we only look at the single variant of this error, I think it's better to use downcasting rather than manually coding all the error variants.

Reply:
Could you expand on that?

Reply:
something like this:

```
// in this crate
struct CargoTomlNotFound;

impl std::error::Error for CargoTomlNotFound {}

// in the lsp-server crate
if let Err(err: Box<dyn std::Error>) = load_workspace() {
   lf let Some(CargoTomlNotFound) = err.downcast_ref() {
   }
}
```

Reply:
@matklad Done!